### PR TITLE
fix: remove dependencies on deprecated auth methods from amplify core

### PR DESCRIPTION
**Pull request by ameter**
*Description of changes:*
fix: remove dependencies on deprecated auth methods from amplify core

*Check points: (check or cross out if not relevant)*

- ~~[ ] Added new tests to cover change, if needed~~
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- ~~[ ] Documentation update for the change if required~~
- [x] PR title conforms to conventional commit style
- ~~[ ] If breaking change, documentation/changelog update with migration instructions~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Amplify/Categories/API/AuthProvider/APIAuthProviderFactory.swift [hunk 1]
```diff
@@ -26,7 +26,11 @@ open class APIAuthProviderFactory {
 
 public protocol AmplifyAuthTokenProvider {
     typealias AuthToken = String
+
+    @available(*, deprecated, message: "Use getUserPoolAccessToken() instead")
```
Comment (brennanMKE) [inline Amplify/Categories/API/AuthProvider/APIAuthProviderFactory.swift:30]: You can use this syntax instead which will let Xcode auto-correct the code.

`@available(*, deprecated, renamed: "getUserPoolAccessToken")`

File: AmplifyPlugins/API/AWSAPICategoryPlugin/Interceptor/RequestInterceptor/AuthTokenProviderWrapper.swift [hunk 2]
```diff
@@ -17,6 +17,7 @@ class AuthTokenProviderWrapper: AuthTokenProvider {
         self.wrappedAuthTokenProvider = tokenAuthProvider
     }
 
+    @available(*, deprecated, message: "Use getUserPoolAccessToken() instead")
```
Comment (brennanMKE) [inline AmplifyPlugins/API/AWSAPICategoryPlugin/Interceptor/RequestInterceptor/AuthTokenProviderWrapper.swift:20]: same with `renamed`

**Review by brennanMKE — COMMENTED** [review]
Left a couple of comments for minor changes.

File: AmplifyPlugins/API/AWSAPICategoryPlugin/Interceptor/RequestInterceptor/AuthTokenProviderWrapper.swift (hunk 2 above)
Comment (ameter) [reply AmplifyPlugins/API/AWSAPICategoryPlugin/Interceptor/RequestInterceptor/AuthTokenProviderWrapper.swift]: done

File: Amplify/Categories/API/AuthProvider/APIAuthProviderFactory.swift (hunk 1 above)
Comment (ameter) [reply Amplify/Categories/API/AuthProvider/APIAuthProviderFactory.swift]: done

**Pull request merged**